service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -169,6 +170,23 @@ public static boolean isCreate(UpdateTableRequest request) {
     return isCreate;
   }
 
+  public ListNamespacesResponse listNamespaces(
+      Namespace parent, String pageToken, Integer pageSize) {
+    PolarisAuthorizableOperation op = PolarisAuthorizableOperation.LIST_NAMESPACES;
+    authorizeBasicNamespaceOperationOrThrow(op, parent);
+
+    if (baseCatalog instanceof IcebergCatalog polarisCatalog) {
+      Page<Namespace> results = polarisCatalog.listNamespaces(parent, pageToken, pageSize);
+      return ListNamespacesResponse.builder()
+          .addAll(results.items)
+          .nextPageToken(results.pageToken.toString())
+          .build();
+    } else {
+      return CatalogHandlers.listNamespaces(
+          namespaceCatalog, parent, pageToken, String.valueOf(pageSize));
```

Comment:
I believe we can do better in Polaris by using the REST API of `baseCatalog` directly.

This can be deferred.

Reply:
It may not be a REST catalog :)

Reply:
Either way, I think we have to delegate pagination to the federated catalog if it supports pagination. `CatalogHandlers.listNamespaces` has a trivial implementation that is sub-optimal when the catalog is a REST Catalog.

Reply:
Followup SGTM